deps/aports/download-repos.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+import tqdm
+import urllib.request
+import urllib.parse
+import tarfile
+import pathlib
+import concurrent.futures
+
+IX_NAME = 'P'
+IX_VERSION = 'V'
+
+def read_index(index):
```

Comment:
`pathlib.Path` exists

Reply:
How does that help here?

Reply:
How? This is the code that parses APKINDEX.tar.gz

Reply:
How about something like this (untested):

```python
def read_index(index):
	index = index.read()
	packages = []
	for block in index.split("\n\n"):
		package = {}
		for line in block.split("\n"):
			key, _, value = line.partition(":")
			package[key] = value
		packages.append(package)
	return packages
```